packages/beacon-node/src/sync/sync.ts

```diff
@@ -114,6 +114,8 @@ export class BeaconSync implements IBeaconSync {
       (currentSlot < 0 ||
         // head is behind clock but close enough with some tolerance
         (headSlot <= currentSlot && headSlot >= currentSlot - this.slotImportTolerance)) &&
+      // If there is no pending checkpoint validation
+      !this.chain.isWSCheckpointValidationPending &&
```

Comment:
Can you explain this in more detail? The logic to consider when the node is synced or not is extremely critical, as it prevents the node from validating. When is this set to true, and when is it set to false? What's the risk of false positives?